rcon/hooks.py

```diff
@@ -238,17 +238,71 @@ def chat_rcon_command(
             )
             return
 
-        commands: dict[str, dict[str, str]] = {}
+        commands: dict[str, dict[str, str | list[str]]] = {}
         for name, params in command.commands.items():
+            logger.info("Original: %s -> %s, args: %s", name, params, args)
             commands[name] = {}
             for k, v in params.items():
-                for i, a in enumerate(args):
-                    v = v.replace(f"${i + 1}", a)
-                commands[name][k] = format_message_string(v, context=ctx)
+                commands[name][k] = replace_params(ctx, args, v)
 
         do_run_commands(rcon, commands)
 
 
+def replace_params(ctx: dict[str, str], args: list[str], v: Any) -> Any:
```

Comment:
Figured I'd quickly comment on why I've changed this from `str or list or dict` to `Any`. First off, using `or` in annotations isn't officially supported syntax (and I doubt many type checkers support it either). Either way, this function accepts more than just those three types, for instance floats and ints, hence `Any`.